src/Microsoft.Azure.CosmosRepository/Repositories/InMemoryRepository.cs

```diff
@@ -61,6 +62,18 @@ internal class InMemoryRepository<TItem> : IRepository<TItem> where TItem : clas
         internal TDeserializeTo DeserializeItem<TDeserializeTo>(string jsonItem) =>
             JsonConvert.DeserializeObject<TDeserializeTo>(jsonItem);
 
+        public async ValueTask<TItem?> TryGetAsync(string id, string? partitionKeyValue = null, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                return await GetAsync(id, partitionKeyValue, cancellationToken);
+            }
+            catch (CosmosException e) when (e.StatusCode is HttpStatusCode.NotFound)
```

Comment:
Are we internally (within the delegated `GetAsync`) throwing a `CosmosException`? If not, we'd probably want to alter this a bit, no?

Reply:
We are throwing that in there, the method `NotFound()` throws one.